api/src/main/resources/workbench.yaml

```diff
@@ -114,6 +114,35 @@ paths:
           description: The user's profile.
           schema:
             $ref: "#/definitions/Profile"
+    patch:
+      tags:
+        - profile
+      consumes:
+        - application/json
+      description: Updates a users profile
+      operationId: updateProfile
+      parameters:
+        - in: body
+          name: updatedProfile
+          description: the new profile to use
+          schema:
+            $ref: "#/definitions/Profile"
+      responses:
+        204:
+          description: Request received.
+        400:
+          description: Bad request
+          schema:
+            $ref: "#/definitions/ErrorResponse"
+    delete:
+      tags:
+        - profile
+      description: Deletes the user's profile and all associated information
```

Comment:
This potentially leaves around a bunch of orphaned Firecloud data. I would definitely not try to address that in this PR but probably worth noting here.

Reply:
Actually, that's a good point. @bopwerks FYI that we can't clean up all information, so it makes sense to make sure usernames are still UUID coded or something so we don't have FireCloud account creation clashes.